OptimizelySDK/Optimizely.cs

```diff
@@ -168,7 +190,8 @@ private bool ValidatePreconditions(Experiment experiment, string userId, UserAtt
         /// <returns>null|Variation Representing variation</returns>
         public Variation Activate(string experimentKey, string userId, UserAttributes userAttributes = null)
         {
-            if (!IsValid)
+            var config = ProjectConfigManager?.GetConfig();
+            if (!IsValid && config == null)
```

Comment:
Should this (and the same check elsewhere) be using an or operator instead of and operator? I.e., `!IsValid || config == null`

Reply:
It should be AND, if isValid is false and also Config is null, then it's invalid.
I will consolidate these conditions into one.

Reply:
`ProjectConfigManager.GetConfig` can return `null`. When it does, we must return `null` early, otherwise attempting to access `config.GetExperimentFromKey` could crash. If I'm misunderstanding, let me know.

Reply:
So `IsValid` will always be `false` if we are using `ProjectConfigManager` to initialize `Optimizely` instance. 
Please let me know we can chat for more clarity.

Reply:
Since we are opening up the `ProjectConfigManager` to be overriden, we should guard against the custom implementation not returning `null` or not setting `isValid` to false. Therefore I think the right apporach is to be more careful and use `!IsValid || config == null` like @mjc1283 suggested.